src/web-components.js

```diff
@@ -48,6 +48,26 @@ export function isShadowDomSupported() {
 }
 
 /**
+ * Returns `true` if Shadow CSS encapsulation is supported.
+ * @return {boolean}
+ */
+export function isShadowCssSupported() {
+  if (!isShadowDomSupported()) {
+    return false;
+  }
+
+  //TODO: Find a better way to test CSS encapsulation
+  if (window.ShadyCSS) {
```

Comment:
This is probably the most interesting part from my perspective: how can we detect and interop with an existing shadowdom polyfill/actual implementation?